src/parquet/util/input.h

```diff
@@ -15,15 +15,63 @@
 // specific language governing permissions and limitations
 // under the License.
 
-#ifndef PARQUET_INPUT_STREAM_H
-#define PARQUET_INPUT_STREAM_H
+#ifndef PARQUET_UTIL_INPUT_H
+#define PARQUET_UTIL_INPUT_H
 
 #include <cstdint>
 #include <memory>
+#include <string>
 #include <vector>
 
 namespace parquet_cpp {
 
+// ----------------------------------------------------------------------
+// Random access input (e.g. file-like)
+
+// Random
+class RandomAccessSource {
+ public:
+  virtual ~RandomAccessSource() {}
+
+  virtual void Close() = 0;
+  virtual size_t Size() = 0;
+  virtual size_t Tell() = 0;
+  virtual void Seek(size_t pos) = 0;
+
+  // Returns actual number of bytes read
+  virtual size_t Read(size_t nbytes, uint8_t* out) = 0;
+};
+
+
+class LocalFileSource : public RandomAccessSource {
+ public:
+  LocalFileSource() : file_(nullptr), is_open_(false) {}
+  virtual ~LocalFileSource();
+
+  void Open(const std::string& path);
+
+  virtual void Close();
+  virtual size_t Size();
+  virtual size_t Tell();
+  virtual void Seek(size_t pos);
+
+  // Returns actual number of bytes read
+  virtual size_t Read(size_t nbytes, uint8_t* out);
+
+  bool is_open() const { return is_open_;}
+  const std::string& path() const { return path_;}
+
+ private:
+  void CloseFile();
+
+  std::string path_;
+  FILE* file_;
+  bool is_open_;
+};
+
+// ----------------------------------------------------------------------
+// Streaming input interfaces
+
```

Comment:
Can you modify the `Peek` and `Read` APIs in `InputStream` and InMemoryInputStreamwith int64_t ?